### PR TITLE
Slack notification sink support.

### PR DESCRIPTION
**Pull request by sureshg**
A native slack notification sink implementation in oneops.

### CODE REVIEW

**Comment by sureshg** [comment]
The [attachment format](https://github.com/oneops/antenna/pull/23/files#diff-86844d7e10956cc5fc77d64fedd752f8R138) is taken from the existing hammock oneops plugin.

```php
  /**
     * Sends messages to the configured slack channel
     *
     * @param $msg message text
     * @return array response map.
     */
    private function sendMessage($msg)
    {
        $level = is_null($msg['severity']) ? "info" : strtolower($msg['severity']);
        $sub = get_subject($msg['subject'], $level);
        $fmt_msg = vsprintf('`%s` | %s | %s', array(
            get_oneops_env($msg['nsPath']),
            ucwords($msg['type']),
            "<" . $msg['notificationUrl'] . " | " . $msg['nsPath'] . ">",
        ));
        $fmt_msg .= "\n" . $msg['text'];
        $attachments = array(array(
            "fallback" => $fmt_msg,
            "pretext" => $sub,
            "color" => get_msg_severity($level),
            "mrkdwn_in" => array("text", "pretext", "fields"),
            "fields" => array(array(
                "title" => '',
                "value" => $fmt_msg,
                "short" => false,))
        ));
        $params = array(
            'text' => '',
            'parse' => 'none',
            'channel' => $this->icfg['channel'],
            'username' => $this->icfg['botname'],
            'attachments' => $attachments,
            // 'icon_url' => $this->iconUrl(48, true),
            // 'icon_emoji' => ':oneops:',
        );
        return $this->postToChannel('', $params);
    }
```

File: src/main/java/com/oneops/antenna/cache/SinkSubscriberLoader.java [hunk 1]
```diff
@@ -141,6 +138,62 @@ public SinkSubscriberLoader(URLSubscriber defaultSystemSubscriber,
...
+     *
+     * @param sink sink object
+     * @return slack subscriber.
+     */
+    private SlackSubscriber buildSlackSub(CmsCI sink) {
+        SlackSubscriber slackSink = new SlackSubscriber();
+        // Channels attribute (Array) format is "team/channel1,team/channel2,...."
+        String[] channels = gson.fromJson(sink.getAttribute("channels").getDfValue(), String[].class);
+        // Text format attribute (Hash) format is "{'pattern|[level]' : 'msg'}, ..."
+        Map<String, String> formats = gson.fromJson(sink.getAttribute("text_formats").getDfValue(),
```
Comment (gauravlall) [inline src/main/java/com/oneops/antenna/cache/SinkSubscriberLoader.java]: hey minor comment gson2.6.2 does not require TypeToken passing Map.class should work

Comment (sureshg) [inline src/main/java/com/oneops/antenna/cache/SinkSubscriberLoader.java]: Using `Map.class` will lose the Generic type params info with `Unchecked Assignment warning Map to Map<String,String>` compiler warning. Anyway, I used it with ` @SuppressWarnings("unchecked")` annotation  as both have the same effect at runtime.

**Comment by gauravlall** [comment]
Looks good 👍

**Pull request merged**